src/Analyzers/Core/Analyzers/SimplifyLinqExpression/SimplifyLinqExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,102 @@
+                       nameOfLocalizableResource: nameof(AnalyzersResources.Simplify_Linq_expression),
+                       resourceManager: AnalyzersResources.ResourceManager,
+                       resourceSource: typeof(AnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterCompilationStartAction(OnCompilationStart);
+
+        private void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            if ((typeof(Enumerable)?.FullName is not string fullyQualifiedName) ||
+                (context.Compilation.GetTypeByMetadataName(fullyQualifiedName) is not INamedTypeSymbol enumerableType))
+            {
+                return;
+            }
+
+            var whereMethodSymbols = enumerableType.GetMembers(nameof(Enumerable.Where)).OfType<IMethodSymbol>().ToImmutableArray();
+            var linqMethodSymbolsBuilder = ArrayBuilder<IMethodSymbol>.GetInstance();
+
+            foreach (var methodName in _nonEnumerableReturningLinqMethodNames)
+            {
+                linqMethodSymbolsBuilder.AddRange(enumerableType.GetMembers(methodName).OfType<IMethodSymbol>());
+            }
+
+            var linqMethodSymbols = linqMethodSymbolsBuilder.ToImmutableAndFree();
+            if (whereMethodSymbols.IsEmpty || linqMethodSymbols.IsEmpty)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                context => AnalyzeInvocationOperation(context, whereMethodSymbols, linqMethodSymbols), OperationKind.Invocation);
+        }
+
+        public void AnalyzeInvocationOperation(OperationAnalysisContext context, ImmutableArray<IMethodSymbol> whereMethods, ImmutableArray<IMethodSymbol> linqMethods)
+        {
+            if (context.Operation.Syntax.GetDiagnostics().Any(diagnostic => diagnostic.Severity == DiagnosticSeverity.Error))
+            {
+                return;
+            }
+
+            if (context.Operation is not IInvocationOperation invocation ||
+                invocation.Arguments.Length != 1 ||
+                invocation.Arguments.Single() is not IArgumentOperation argument)
+            {
+                return;
+            }
+
+            if (linqMethods.Any(m => m.Equals(invocation.TargetMethod.OriginalDefinition, SymbolEqualityComparer.Default)) &&
+                argument.Children.FirstOrDefault() is IInvocationOperation whereInvocation &&
+                whereMethods.Any(m => m.Equals(whereInvocation.TargetMethod.OriginalDefinition, SymbolEqualityComparer.Default)))
+            {
+                context.ReportDiagnostic(
+                    DiagnosticHelper.Create(
+                        Descriptor,
+                        invocation.Syntax.GetLocation(),
```

Comment:
i tink you'll have to do some syntactic checks here to make sure the syntax is as you expect.  otherwise, you don't really know what's going to happen on the fixer side, and you can def crash.